tests/e2e/test_cd.py

```
@@ -81,3 +81,77 @@
+
+def test_cd_error_permission_denied(shell_session):
+    if os.path.exists("/root"):
+        shell_session.sendline("cd /root")
+        shell_session.expect(PROMPT)
+        shell_session.sendline("echo $?")
+        shell_session.expect(PROMPT)
+        result = get_command_output(shell_session.before)
+        assert result == "1"
+
+
+def test_cd_error_permission_denied(shell_session):
```

Ruff

Redefinition of unused `test_cd_error_permission_denied` from line 128

(F811)